### ISSUE TITLE
Add support for name generators on the server side

### ISSUE THREAD

**Issue by ivantopo** [kamon-io/kamon-instrumentation-common#2]
Basically add support for the very same generators that we have on the client side but for the server side as well. This is specially important for people migrating from Kamon 1.x and using Akka HTTP because we did have support for name generators on the server side and we don't have it anymore.

/cc @pnerg @TimSoethout

**Comment by pnerg** [comment]
Any ETA on this? It's kind of blocking our migration to kamon 2.x.

I've looked into the `kamon.tracing.operations-mappings` config but it won't help us as it only "masks" identifiers, I actually need some of these when building the operation name.
E.g.
```
/api/<addon>/<resource>/<id> -> <resource>
/api/<addon/<resource>/<id>/<relationship> -> <resource>-<relationship>
```
So I need to decompose the path into parts and use some of them into the operation name.
The only way is to be able to provide a name generator in code

**Comment by l15k4** [comment]
The #3 is not a fix, right? I'm just trying to migrate kamon 1.x -> 2.x while knowing very little about it :-) But the `HttpOperationNameGenerator` has quite different interface...

**Comment by pnerg** [comment]
It's the same interface, the difference is in the [config](https://github.com/pnerg/area51-kamon/blob/master/akka-http/src/main/resources/kamon.conf#L251-L254) where one needs to configure client/server name generator separately.

The code for the client/server generators is [here](https://github.com/pnerg/area51-kamon/blob/master/common/src/main/scala/org/dmonix/area51/kamon/NameGenerators.scala)

**Comment by l15k4** [comment]
I see, that's good news, thank you @pnerg ... this issue could be closed perhaps then?

**Comment by pnerg** [comment]
Yeah should probably be as my PR solved parts of this ticket and then I believe this [commit](https://github.com/kamon-io/kamon-instrumentation-common/commit/1f6712cd308f4d65411704c358719aaccf2ab6ef) solved the remaining issue where the operation name got lost/overwritten

**Comment by l15k4** [comment]
Oki, looks good :+1:  Please close
